### DISCUSSION TITLE
1.2.0 Beta 9 - Beta 15: Buffer-related distortion

### DISCUSSION THREAD

**Discussion by sklogw0** [apohl79/audiogridder#1092] [Help]
**Use case 01:** Windows AAX in Pro Tools - Main machine - plugin, laptop - server; Flawless signal at any block size. Clean audio stream with or without plugins served. **_This has been 99% of my experience with this wonderful software from the first day of AAX implementation._**

**Use case 02:** Windows VST in FL20 - Laptop - plugin, main machine - server; Signal doubled at small latency with heavy distortion in collapsed mono on top of original stereo signal, with or without plugins served. Clean signal only possible with no buffer in local mode, with or without plugins served. Using FL menu plugin specific settings, fixed buffer options offer no change or make it worse depending on setting. 

Tried updating from AG Beta 9 to AG Beta 15, using fixed buffer or plugin specific buffer options offers no change or made it worse depending on setting. **_I am one week into this use case._** Is this operator error or something specific to FL20's audio engine and VST handling?

**Use case 03:** Windows VST in FL20 - Main machine - plugin, laptop - server; Same outcome as **Use case 02**.

![FL20AudioSettings](https://user-images.githubusercontent.com/117948912/201255715-edbd7109-7148-4513-8151-1595091a4910.png)

**Comment by emack999** [comment]
Do you have WiFi enabled?  If so, you won't get good results.  Make sure you disable WiFi and use LAN connection.

**Reply by sklogw0** [reply]
Negative. Hardlines and dedicated IPs.

**Reply by sklogw0** [reply]
**Use case  01** would exhibit the same problem if the issue was network bound. 

There are two variables here, AAX vs VST as a platforn and how they differ in handling audio streams, and Pro Tools vs. FL20 with the differences in their audio engines. AG might need to be accounting for something specific to FL and how it handles VST compared to other DAWs.

**Comment by sklogw0** [comment]
Photo uploaded and added to OP to assist with reproducing the problem.

**Reply by apohl79** [reply]
Just to clarify, this issues started with beta9 and versions prior to that were doing ok?

**Reply by sklogw0** [reply]
I can't speak to that. I've exclusively been using AG with Pro Tools since the first day of your AAX implementation. I've updated periodically as your optimizations netted greater outcomes. I was simply at 1.2.0 Beta 9 with no issues and no need for the new features you've added in 10 forward. Then about a week ago I've started needing AG VST for load distribution in FL20. The problems in OP-defined **Use case 02** and **Use case 03** are what I encountered. I tried the updates through Beta 15, same results.

**Reply by apohl79** [reply]
ok understood.

**Reply by apohl79** [reply]
Another question: Is this problem happening with and without the fixed size buffer setting in FL?

<img width="568" alt="image" src="https://user-images.githubusercontent.com/1129160/201298635-831f7061-f9a4-4e3d-9ced-404855e4222e.png">

**Reply by apohl79** [reply]
Ah I think you tried them all, right?

**Reply by sklogw0** [reply]
Yes, those were off by default but I tried enabling 'Use fixed buffers' and toggling both options, "process maximum size buffer" and "use maximum buffer size from host" in the "More" sub menu. No change or worse results.

**Reply by apohl79** [reply]
And within AG you have also tried fixed buffers on and off?

**Reply by sklogw0** [reply]
> And within AG you have also tried fixed buffers on and off?

Enabling AG Beta 15's "use fixed size outbound buffer" introduces indecipherable stuttering at lengths that appear to be determined/influenced by block size, with or without plugins served. Disabled, it's still as defined in OP.

Enabling AG Beta 15's 'allow individual buffer size by plugin' makes it less responsive, audible changes but it's at a point I can't tell if it's because of the setting or something else because it's not a clean or smooth stream from the start before AG even has a plugin selected and served.
